src/coreclr/nativeaot/System.Private.CoreLib/src/Internal/Runtime/CompilerHelpers/InteropHelpers.cs

```diff
@@ -341,7 +341,10 @@ internal static unsafe void FixupModuleCell(ModuleFixupCell* pCell)
                 {
                     // Built-in rules didn't resolve the library. Use AssemblyLoadContext as a last chance attempt.
                     AssemblyLoadContext? loadContext = AssemblyLoadContext.GetLoadContext(callingAssembly);
-                    hModule = loadContext.GetResolvedUnmanagedDll(callingAssembly, moduleName);
+                    if (loadContext != null)
```

Comment:
`GetLoadContext` will only return null if the passed in assembly is not a runtime-implemented assembly. This code should crash if that happens because that's a bug and we shouldn't try to paper over it.

Reply:
In that specific case, couple lines below it would be handled by 
```
// If the module is still unresolved, this is an error.
loadLibErrorTracker.Throw(moduleName);
```
So that's seems to correct skipping.

Reply:
That exception blames the wrong thing though - it sends whoever hits it the wrong way. Code that fails fast is easier to work with than code that makes bugs take a scenic route through the codebase before they manifest themselves.

Reply:
Make sense. Here I thought that this is proper way to handle.